delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,510 @@
+# Copyright 2013-2019 Yannick Vaucher (Camptocamp SA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import json
+import logging
+import re
+import threading
+import urllib.parse
+from datetime import datetime, timedelta
+from io import BytesIO
+
+from odoo import _, exceptions
+
+import requests
+from PIL import Image
+
+_logger = logging.getLogger(__name__)
+
+_compile_itemid = re.compile(r"[^0-9A-Za-z+\-_]")
+_compile_itemnum = re.compile(r"[^0-9]")
+AUTH_PATH = "/WEDECOAuth/token"
+GENERATE_LABEL_PATH = "/api/barcode/v1/generateAddressLabel"
+
+
+class PostlogisticsWebService(object):
+
+    """ Connector with PostLogistics for labels using post.ch API
+
+    Handbook available here:
+    https://developer.post.ch/en/digital-commerce-api
+    https://wedec.post.ch/doc/swagger/index.html?
+        url=https://wedec.post.ch/doc/api/barcode/v1/swagger.yaml
+        #/Barcode/generateAddressLabel
+
+    Allows to generate labels
+
+    """
+
+    access_token = False
+    access_token_expiry = False
+    _lock = threading.Lock()
+
+    def __init__(self, company):
+        self.default_lang = company.partner_id.lang or "en"
+
+    def _get_language(self, lang):
+        """ Return a language to iso format from odoo format.
+
+        `iso_code` field in res.lang is not mandatory thus not always set.
+        Use partner language if available, otherwise use english
+
+        :param partner: partner browse record
+        :return: language code to use.
+
+        """
+        if not lang:
+            lang = self.default_lang
+        available_languages = ["de", "en", "fr", "it"]  # Given by API doc
+        lang_code = lang.split("_")[0]
+        if lang_code in available_languages:
+            return lang_code
+        return "en"
+
+    def _prepare_recipient(self, picking):
+        """ Create a ns0:Recipient as a dict from a partner
+
+        :param partner: partner browse record
+        :return a dict containing data for ns0:Recipient
+
+        """
+        partner = picking.partner_id
+        partner_mobile = picking.delivery_mobile or partner.mobile
+        partner_phone = picking.delivery_phone or partner.phone
+
+        if partner.postlogistics_notification == "email" and not partner.email:
+            raise exceptions.UserError(_("Email is required for notification."))
+        elif partner.postlogistics_notification == "sms" and not partner_mobile:
+            raise exceptions.UserError(
+                _("Mobile number is required for sms notification.")
+            )
+        elif partner.postlogistics_notification == "phone" and not partner_phone:
+            raise exceptions.UserError(
+                _("Phone number is required for phone call notification.")
+            )
+
+        partner_name = partner.name or partner.parent_id.name
+        recipient = {
+            "name1": partner_name,
```

Comment:
```suggestion
            "name1": partner_name[:35],
```

Reply:
Added in https://github.com/OCA/delivery-carrier/pull/323/commits/29e751d5d5865b752a91ef4195bc06ae167d944f, thanks